node/service/src/parachains_db/upgrade.rs

```diff
@@ -102,3 +128,131 @@ fn migrate_from_version_0_to_1(path: &Path) -> Result<(), Error> {
+				if changed {
+					gum::debug!(
+						target: LOG_TARGET,
+						"Column {} will be cleared. Old options: {:?}, New options: {:?}",
+						idx,
+						opts,
+						options.columns[idx]
+					);
+					Some(idx)
+				} else {
+					None
+				}
+			})
+			.collect::<Vec<_>>();
+
+		if columns_to_clear.len() > 0 {
+			gum::debug!(
+				target: LOG_TARGET,
+				"Database column changes detected, need to cleanup {} columns.",
+				columns_to_clear.len()
+			);
+		}
+
+		for column in columns_to_clear {
+			gum::debug!(target: LOG_TARGET, "Clearing column {}", column,);
+			parity_db::clear_column(path, column.try_into().expect("Invalid column ID"))
+				.map_err(|e| other_io_error(format!("Error clearing column {:?}", e)))?;
+		}
+
+		// Write the updated column options.
+		options
+			.write_metadata(path, &metadata.salt)
+			.map_err(|e| other_io_error(format!("Error writing metadata {:?}", e)))?;
+	}
+
+	Ok(())
+}
+
+/// Database configuration for version 1.
+pub(crate) fn paritydb_version_1_config(path: &Path) -> parity_db::Options {
+	let mut options =
+		parity_db::Options::with_columns(&path, super::columns::v1::NUM_COLUMNS as u8);
+	for i in columns::v1::ORDERED_COL {
+		options.columns[*i as usize].btree_index = true;
+	}
+
+	options
+}
+
+/// Database configuration for version 0. This is useful just for testing.
+#[cfg(test)]
+pub(crate) fn paritydb_version_0_config(path: &Path) -> parity_db::Options {
+	let mut options =
+		parity_db::Options::with_columns(&path, super::columns::v1::NUM_COLUMNS as u8);
+	options.columns[super::columns::v1::COL_AVAILABILITY_META as usize].btree_index = true;
+	options.columns[super::columns::v1::COL_CHAIN_SELECTION_DATA as usize].btree_index = true;
+
+	options
+}
+
+/// Migration from version 0 to version 1.
+/// Cases covered:
+/// - upgrading from v0.9.23 or earlier -> the `dispute coordinator column` was changed
+/// - upgrading from v0.9.24+ -> this is a no op assuming the DB has been manually fixed as per
+/// release notes
+fn paritydb_migrate_from_version_0_to_1(path: &Path) -> Result<(), Error> {
+	// Delete the `dispute coordinator` column if needed (if column configuration is changed).
+	paritydb_fix_columns(
+		path,
+		paritydb_version_1_config(path),
+		vec![super::columns::v1::COL_DISPUTE_COORDINATOR_DATA],
+	)?;
+
+	Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+	#[test]
+	fn test_paritydb_migrate_0_1() {
+		use super::{columns::v1::*, *};
+		use parity_db::Db;
+
+		let db_dir = tempfile::tempdir().unwrap();
+		let path = db_dir.path();
+		{
+			let db = Db::open_or_create(&paritydb_version_0_config(&path)).unwrap();
+
+			db.commit(vec![
+				(COL_DISPUTE_COORDINATOR_DATA as u8, b"1234".to_vec(), Some(b"somevalue".to_vec())),
+				(COL_AVAILABILITY_META as u8, b"5678".to_vec(), Some(b"somevalue".to_vec())),
+			])
+			.unwrap();
+		}
+
+		try_upgrade_db(&path, DatabaseKind::ParityDB).unwrap();
+
+		let db = Db::open(&paritydb_version_1_config(&path)).unwrap();
```

Comment:
Thanks for adding a test. Can we use `open_creating_paritydb` instead?

Reply:
Actually, might be better idea to leave as is. This tests opening with v1, not the current version.